lms/templates/dashboard.html

```diff
@@ -108,28 +113,61 @@
-              <% show_email_settings = (enrollment.course_id in show_email_settings_for) %>
-              <% course_mode_info = all_course_modes.get(enrollment.course_id) %>
-              <% is_paid_course = (enrollment.course_id in enrolled_courses_either_paid) %>
-              <% is_course_blocked = (enrollment.course_id in block_courses) %>
-              <% course_verification_status = verification_status_by_course.get(enrollment.course_id, {}) %>
-              <% course_requirements = courses_requirements_not_met.get(enrollment.course_id) %>
-              <% related_programs = inverted_programs.get(unicode(enrollment.course_id)) %>
-              <% show_consent_link = (enrollment.course_id in consent_required_courses) %>
-              <%include file='dashboard/_dashboard_course_listing.html' args='course_overview=enrollment.course_overview, enrollment=enrollment, show_courseware_link=show_courseware_link, cert_status=cert_status, can_unenroll=can_unenroll, credit_status=credit_status, show_email_settings=show_email_settings, course_mode_info=course_mode_info, is_paid_course=is_paid_course, is_course_blocked=is_course_blocked, verification_status=course_verification_status, course_requirements=course_requirements, dashboard_index=dashboard_index, share_settings=share_settings, user=user, related_programs=related_programs, display_course_modes_on_dashboard=display_course_modes_on_dashboard, show_consent_link=show_consent_link, enterprise_customer_name=enterprise_customer_name' />
+            % for dashboard_index, enrollment in enumerate(course_entitlements + course_enrollments):
+              <%
+                # Check if the course run is an entitlement and if it has an associated session
+                entitlement = enrollment if isinstance(enrollment, CourseEntitlement) else None
+                entitlement_session = entitlement.enrollment_course_run if entitlement else None
+
+                is_fulfilled_entitlement = True if entitlement and entitlement_session else False
+                is_unfulfilled_entitlement = True if entitlement and not entitlement_session else False
+
+                entitlement_available_sessions = []
+                if entitlement:
+                  # Grab the available, enrollable sessions for a given entitlement and scrape them for relevant attributes
+                  entitlement_available_sessions = [{
+                    'session_id': course['key'],
+                    'enrollment_end': course['enrollment_end'],
+                    'pacing_type': course['pacing_type'],
+                    'session_start_advertised': CourseOverview.get_from_id(CourseKey.from_string(course['key'])).advertised_start,
+                    'session_start': CourseOverview.get_from_id(CourseKey.from_string(course['key'])).start,
+                    'session_end': CourseOverview.get_from_id(CourseKey.from_string(course['key'])).end,
```

Comment:
This is even more expensive if we're making the call to get the CourseOverview 3 times for each course run on each entitlement. If CourseOverview is mongo-based, I'm not sure if the query would be cached or not on the backend. We should, at the very least, cache it within this method by saving it to a variable and reuse the same object for each field that needs to be serialized.